sqlalchemylogger/handlers.py

```diff
@@ -1,42 +1,109 @@
 import logging
 import traceback
+import sqlalchemy
 from sqlalchemy import create_engine
 from sqlalchemy.orm import sessionmaker
-from .models import Log, Base
+from .models import createLogClass, Base
 from sqlalchemy.exc import OperationalError, InvalidRequestError
+from sqlalchemy_utils import database_exists, create_database
+import threading, queue, time
+from .filters import ContainsExpression, DoesNotContainExpression
+
+
+module_logs = logging.getLogger(__name__)
 
 class SQLAlchemyHandler(logging.Handler):
-    def __init__(self, sqlalchemyUrl):
+    
+    MAX_NB_LOGS = 100
+    MAX_TIMEOUT = 1
+
+    def __init__(self, sqlalchemyUrl, 
+              doesNotContainExpression = None,
+              containsExpression = None):
         super().__init__()
-        self.engine = create_engine(sqlalchemyUrl)
+        # initialize DB session
+        self.engine = create_engine(sqlalchemyUrl['url'])
+        self.Log = createLogClass(
+                  tablename = sqlalchemyUrl.get('tablename', 'logs'),
+                  tableargs = sqlalchemyUrl.get('tableargs', None))
         Base.metadata.bind = self.engine
         DBSession = sessionmaker(bind=self.engine)
         self.session = DBSession()
+        # initialize log queue
+        self.log_queue = queue.Queue()
+        # initialize a thread to process the logs Asynchronously
+        self.processor_thread = threading.Thread(target = self._processor, daemon = True)
+        self.processor_thread.start()
+        # initialize filters
+        if doesNotContainExpression:
+             self.addFilter(DoesNotContainExpression(doesNotContainExpression))
+        if containsExpression:
+             self.addFilter(ContainsExpression(containsExpression))
+
+
+    def _processor(self):
```

Comment:
I would do something like that, instead:
```python
from threading import Condition
import logging


class SQLAlchemyHandler(logging.Handler):
    def __init__(self):
        self.condition = Condition()
        self.queue = []
        # init the rest

    def _processor(self):
        while True:
            with self.condition:
                while not self.queue:
                    self.condition.wait()
                queue = self.queue
                self.queue = []
            self._write_logs(queue)

    def emit(self, record):
        log = self.Log(...)
        with self.condition:
            self.queue.append(log)
            self.condition.notify()
```

Reply:
why do you dislike the usage of proper `queue.Queue()` ? does it add too much overhead?

Reply:
It's more code...

Reply:
https://github.com/danduk82/poc_oereb_logger/pull/4/commits/d20a0a2e66e76360e7fba31200f9659dd576d493